src/ForeignCallbacks.jl

```diff
@@ -8,38 +8,34 @@ struct Node{T}
 end
 Node(data::T) where T = Node{T}(C_NULL, data)
 
+setnext!(node::Ptr{Node{T}}, next::Ptr{Node{T}}) where {T} =
+    unsafe_store!(convert(Ptr{Ptr{Node{T}}}, node), next)
+
 function calloc(::Type{T}) where T 
     ptr = Libc.malloc(sizeof(T))
     ccall(:memset, Ptr{Cvoid}, (Ptr{Cvoid}, Cint, Csize_t), ptr, 0, sizeof(T))
     return convert(Ptr{T}, ptr)
 end
 
-mutable struct LockfreeQueue{T}
-    @atomic head::Ptr{Node{T}}
-    @atomic tail::Ptr{Node{T}}
-    function LockfreeQueue{T}() where T
-        @assert !Base.ismutabletype(T) && Base.isconcretetype(T) && Base.datatype_pointerfree(T)
-        tmp = calloc(Node{T})
-        new{T}(tmp, tmp)
+mutable struct Stack{T}
+    @atomic top::Ptr{Node{T}}
+    function Stack{T}() where T
+        @assert Base.datatype_pointerfree(Some{T})
```

Comment:
Why `Some{T}`

Reply:
If I put `T` in an immutable struct, I can check `!ismutabletype` and `datatype_pointerfree` in one go. I also don't need `isconcretetype` this way. I thought it's also slightly more general since it allows things like `T = Union{Int,Nothing}`.

Reply:
Maybe `Base.datatype_pointerfree(Node{T})` is better since that's what we actually use

Reply:
Not that I'm planning to use it this way though. So I'm totally OK with rolling it back. I just thought it's neat :)

Reply:
I would be ok with `Node{T}, this was weird reading the code.